ethminer/MinerAux.h

```diff
@@ -855,19 +854,6 @@ class MinerCLI
 					}
 					//exit(0);
 				}
-				else if (EthashAux::eval(previous.seedHash, previous.headerHash, solution.nonce).value < previous.boundary)
```

Comment:
It was to support the second of two work packages: `current.headerHash` and `previous.headerHash`, where `previous` is the "stale" block. This was to check if a solution was for the `previous` block, there was an issue about it here: https://github.com/Genoil/cpp-ethereum/issues/36 ([claims](https://github.com/Genoil/cpp-ethereum/issues/36#issuecomment-218005606) to be a 1.2% improvement).

When new work updates are coming very frequently (e.g. adding new tx's to the pending block every 1 second), sometimes a solution to `previous` is found just as a new one is arriving. But when the new work package arrives, the one in `current` becomes `previous`, and the one in `previous` is purged. So keeping around only the last two work packages is not enough to prevent throwing away solutions.

Rather than keep around three work packages to check solutions against, the solutions themselves now contain the work package information (seedHash, headerHash, nonce, boundary). So now it just checks that the solution is valid before submitting it to the client (`eth_submitWork`), rather than checking if its a solution to the current (or previous) work package.